### PR TITLE
[KotlinCleanup] ActivityExportingDelegate.kt

### PR DESCRIPTION
**Pull request by Jatin887**
## Purpose / Description
Kotlin CleanUp

## Fixes
Part of #10489
## Approach
Using lint and some code changes which were mentioned in the comments and required to make the code more reliable
## How Has This Been Tested?

Physical Device One Plus Nord2

## Checklist
_Please, go through these checks before submitting the PR._

- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/anki/dialogs/ExportCompleteDialog.kt [hunk 1]
```diff
@@ -50,11 +50,11 @@ class ExportCompleteDialog(private val listener: ExportCompleteDialogListener) :
             .negativeText(R.string.export_save_button)
             .onPositive { _: MaterialDialog?, _: DialogAction? ->
                 listener.dismissAllDialogFragments()
-                listener.emailFile(exportPath)
+                listener.emailFile(exportPath!!)
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/anki/dialogs/ExportCompleteDialog.kt:53]: maybe move the `!!` to the variable declaration

File: AnkiDroid/src/main/java/com/ichi2/anki/export/ActivityExportingDelegate.kt [hunk 2]
```diff
@@ -150,14 +145,13 @@ class ActivityExportingDelegate(private val activity: AnkiActivity, private val
...
-        val isSuccessful = exportToProvider(result.data, true)
+        val isSuccessful = true.exportToProvider(result.data!!)
         @StringRes val message = if (isSuccessful) R.string.export_save_apkg_successful else R.string.export_save_apkg_unsuccessful
         showSimpleSnackbar(activity, activity.getString(message), isSuccessful)
     }
 
-    @KotlinCleanup("make intent non-null")
-    private fun exportToProvider(intent: Intent?, deleteAfterExport: Boolean): Boolean {
-        if (intent == null || intent.data == null) {
+    private fun Boolean.exportToProvider(intent: Intent): Boolean {
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/anki/export/ActivityExportingDelegate.kt:153]: I don't think this should be an extension

**Review by david-allison — CHANGES_REQUESTED** [review]

File: AnkiDroid/src/main/java/com/ichi2/anki/dialogs/ExportCompleteDialog.kt (hunk 1 above)
Comment (Jatin887) [reply AnkiDroid/src/main/java/com/ichi2/anki/dialogs/ExportCompleteDialog.kt:53]: image
is it ok?

Comment (david-allison) [reply AnkiDroid/src/main/java/com/ichi2/anki/dialogs/ExportCompleteDialog.kt:53]: yes

File: AnkiDroid/src/main/java/com/ichi2/anki/export/ActivityExportingDelegate.kt (hunk 2 above)
Comment (Jatin887) [reply AnkiDroid/src/main/java/com/ichi2/anki/export/ActivityExportingDelegate.kt]: changes added

**Review by mikehardy — APPROVED** [review]
nice!

**Pull request merged**